clients/drcachesim/analyzer.cpp

```diff
@@ -339,6 +339,14 @@ analyzer_tmpl_t<RecordType, ReaderType>::init_scheduler_common(
             uint64_t filetype = scheduler_.get_stream(i)->get_filetype();
             VPRINT(this, 2, "Worker %d filetype %" PRIx64 "\n", i, filetype);
             if (TESTANY(OFFLINE_FILE_TYPE_CORE_SHARDED, filetype)) {
+                if (i == 0 && shard_type_ == SHARD_BY_CORE) {
```

Comment:
This check is certainly useful. It can be added without preferred_shard_type too right?

Reply:
Yes, but it does fit with the rest of the PR so seems fine to leave it in this PR.